src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/androids/menu/AndroidShareMenu.java

```diff
@@ -0,0 +1,269 @@
+        Validate.notNull(users, "The trusted users list cannot be null!");
+
+        if (users.contains(p.getUniqueId().toString())) {
+            users.remove(p.getUniqueId().toString());
+            SlimefunPlugin.getLocalization().sendMessage(owner, "android.access-manager.messages.delete-success", msg -> msg.replace("%player%", p.getName()));
+
+            setValue(android.getState(), users.toString());
+        } else {
+            SlimefunPlugin.getLocalization().sendMessage(owner, "android.access-manager.messages.is-not-trusted-player", msg -> msg.replace("%player%", p.getName()));
+        }
+    }
+
+    /**
+     * Parse trusted player list raw string to List.
+     *
+     * @param value list raw string
+     * @return trusted player list
+     */
+    private @Nonnull static List<String> parseBlockInfoToList(@Nonnull String value) {
+        Validate.notNull(value, "The trusted player list cannot be null!");
+
+        String replacedText = value.replace("[", "").replace("]", "");
+
+        if (replacedText.isEmpty()) {
+            return Collections.emptyList();
+        } else {
+            return Arrays.asList(replacedText.split(", "));
+        }
+    }
+
+    /**
+     * Get a trusted users list of specific android.
+     *
+     * @param b the block of a Android
+     * @return Trusted users
+     */
+    public @Nonnull static List<String> getTrustedUsers(@Nonnull Block b) {
+        Validate.notNull(b, "The android block cannot be null!");
+
+        Optional<String> trustUsers = getValue(b.getState());
+
+        // Checks for old Android
+        if (!trustUsers.isPresent()) {
+            setValue(b.getState(), Collections.emptyList().toString());
+            return Collections.emptyList();
+        }
+
+        return parseBlockInfoToList(trustUsers.get());
+    }
+
+    /**
+     * Checks user is in trusted users list.
+     *
+     * @param b the block of a Android
+     * @param uuid user's UUID
+     * @return whether is the trusted user of android or not
+     */
+    @ParametersAreNonnullByDefault
+    public static boolean isTrustedUser(Block b, UUID uuid) {
+        Validate.notNull(b, "The android block cannot be null!");
+        Validate.notNull(uuid, "The UUID of player to check cannot be null!");
+
+        Optional<String> trustUsers = getValue(b.getState());
+
+        return trustUsers.map(s -> s.contains(uuid.toString())).orElse(false);
+    }
+
+    private static void setValue(@Nonnull BlockState state, @Nonnull String value) {
+        Validate.notNull(state, "The android block state cannot be null!");
+        Validate.notNull(value, "The data value cannot be null!");
+
+        if (!(state instanceof TileState)) {
+            return;
+        }
+
+        try {
+            PersistentDataContainer container = ((TileState) state).getPersistentDataContainer();
+            container.set(BLOCK_INFO_KEY, PersistentDataType.STRING, value);
```

Comment:
In CS-CoreLib2 there is a special class for storing UUIDs. It is called `PersistentUUIDDataType`.

Reply:
It seems only support store single UUID? it implements PersistentDataType<int[], UUID>

Reply:
Oh, because you are serializating those UUIDs. Okay, my mistake.